yt/data_objects/particle_trajectories.py

```diff
@@ -111,8 +115,10 @@ def __init__(
             mylog.setLevel(old_level)
 
         sorted_storage = sorted(my_storage.items())
-        times = [time for _fn, (time, *_) in sorted_storage]
-        self.times = self.data_series[0].arr(times, times[0].units)
+        times = [time.to("Myr") for _fn, (time, *_) in sorted_storage]
+        self.times = self.data_series[0].arr(
+            [time.value for time in times], times[0].units
+        )
```

Comment:
I'm guessing you force-convert times to `"Myr"` as a way to fix a bug where `self.times` would be incorrectly constructed from a heterogenous array of times. I would however favour harmonising the unit to `times[0].units` instead of hardcoding `Myr` which looks like a severe "yt-ism" to me and wouldn't make much sense outside of astronomy.